### DISCUSSION TITLE
Get Memory Used By WASM

### DISCUSSION THREAD

**Discussion by StaticallyTypedAnxiety** [WasmEdge/WasmEdge#2347] [Q&A] [answered]
Hello,

I was wondering, what is the best way to get the not the maximum but the current used memory by the wasm sandbox? Preferably in rust.

Thanks!

**Comment by hydai** [comment, marked as answer]
Yes, please refer to this API: https://github.com/WasmEdge/WasmEdge/blob/master/include/api/wasmedge/wasmedge.h#L2431-L2437

**Reply by StaticallyTypedAnxiety** [reply]
Hello, thank you for the quick response.  I think the problem with this one is that it ceilings it off to the nearest page. Is there anyway to get something more exact than that? Thank you

**Reply by hydai** [reply]
Obtaining an exact memory space in WebAssembly (wasm) is unfeasible because additional memory pages must be allocated as needed. Once a page is used, the memory space utilized is already allocated and marked as in use.
